### PR TITLE
No issue: Partially update the add-on manager list.

### PR DESCRIPTION
**Pull request by Amejia481**
Closes https://github.com/mozilla-mobile/fenix/issues/10245 and https://github.com/mozilla-mobile/fenix/issues/10246
[Fenix Before](https://drive.google.com/file/d/1Mg7XJTFTIaP_qG5ZKvmUvWtNWdVnlMPj/view?usp=sharing)
[Fenix Now](https://drive.google.com/file/d/1lDANVgMAt5yETkYj3MROhj6hVWBosx3N/view?usp=sharing) 

### Pull Request checklist

- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features. In addition, it includes a screenshot of a successful [accessibility scan](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor&hl=en_US) to ensure no new defects are added to the product.

### After merge
- [ ] **Milestone**: Make sure issues finished by this pull request are added to the [milestone](https://github.com/mozilla-mobile/fenix/milestones) of the version currently in development.

### To download an APK when reviewing a PR:
1. click on Show All Checks,
2. click Details next to "Taskcluster (pull_request)" after it appears and then finishes with a green checkmark,
3. click on the "Fenix - assemble" task, then click "Run Artifacts".
4. the APK links should be on the left side of the screen, named for each CPU architecture

### CODE REVIEW

File: app/src/main/java/org/mozilla/fenix/addons/AddonsManagementFragment.kt [hunk 1]
```diff
@@ -63,20 +60,6 @@ class AddonsManagementFragment : Fragment(R.layout.fragment_add_ons_management),
         findPreviousDialogFragment()?.let { dialog ->
             dialog.onPositiveButtonClicked = onPositiveButtonClicked
         }
-
-        scope = requireContext().components.core.store.flowScoped { flow ->
-            flow.ifChanged { it.extensions }
-                .collect { _ ->
-                    view?.let {
-                        bindRecyclerView(it)
```
Comment (Amejia481) [inline app/src/main/java/org/mozilla/fenix/addons/AddonsManagementFragment.kt:71]: I found we were refreshing the whole list multiple times. When we install an add-on, we are getting notified 
one times on `InstallWebExtensionAction` and two times on `UpdateBrowserAction`

File: app/src/main/java/org/mozilla/fenix/addons/AddonsManagementFragment.kt [hunk 2]
```diff
@@ -210,7 +199,7 @@ class AddonsManagementFragment : Fragment(R.layout.fragment_add_ons_management),
                             it.translatedName
                         )
                     )
-                    bindRecyclerView(view)
+                    adapter?.refresh(it)
```
Comment (Amejia481) [inline app/src/main/java/org/mozilla/fenix/addons/AddonsManagementFragment.kt:202]: When we are installing the add-on, we exactly which add-on has changed

**Comment by ekager** [comment]
Failing on: 
> [task 2020-04-27T16:05:12.307Z] e: /builds/worker/checkouts/src/app/src/test/java/org/mozilla/fenix/components/metrics/BreadcrumbRecorderTest.kt: (22, 26): Return type of 'report' is not a subtype of the return type of the overridden member 'public abstract fun report(throwable: Throwable): String? defined in mozilla.components.lib.crash.service.CrashReporterService'
> [task 2020-04-27T16:05:12.307Z] e: /builds/worker/checkouts/src/app/src/test/java/org/mozilla/fenix/components/metrics/BreadcrumbRecorderTest.kt: (25, 26): Return type of 'report' is not a subtype of the return type of the overridden member 'public abstract fun report(crash: Crash.UncaughtExceptionCrash): String? defined in mozilla.components.lib.crash.service.CrashReporterService'
> [task 2020-04-27T16:05:12.308Z] e: /builds/worker/checkouts/src/app/src/test/java/org/mozilla/fenix/components/metrics/BreadcrumbRecorderTest.kt: (28, 26): Return type of 'report' is not a subtype of the return type of the overridden member 'public abstract fun report(crash: Crash.NativeCodeCrash): String? defined in mozilla.components.lib.crash.service.CrashReporterService'
> [task 2020-04-27T16:05:12.308Z]

**Comment by Amejia481** [comment]
It should be fixed now :)

**Pull request merged**